docs/reference/release-notes/highlights-7.7.0.asciidoc

```diff
@@ -18,3 +18,110 @@ pivot and summarize your data and store it in a new index. See
+returning the final response only once it's entirely available, it will pan the
+results as they become available, asynchronously.
+
+The request takes a parameter, `wait_for_completion`, which controls how long
+the server will wait until it sends back a response. The first response
+contains among others a search unique ID, a response version, an indication if
+this response is partial or not, plus the usual metadata (shards involved,
+number of hits etc) and potentially results. If the response is not complete
+and final, the client can continue polling for results issuing a new request
+using the provided search ID; if new results are available, the returned
+version is incremented and the new results batch returned. This can go on until
+all the results are fetched.
+
+Unless deleted earlier by the user, the asynchronous searches are kept alive
+for a given interval. This defaults to 5 days and can be controlled by another
+request parameter, `keep_alive`.
+
+This functionality is offered under the free {xpack} Basic license.
+
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[float]
+=== Password protection for the keystore
+
+{es} uses a custom on-disk keystore for secure settings such as
+passwords and SSL certificates. Up until now, this prevented users with
+command-line access from viewing secure files by listing commands, but nothing
+prevented such users from changing values in the keystore, or removing values
+from it. Furthermore, the values were only obfuscated by a hash; no
+user-specific secret protected the secure settings.
+
+This new feature changes all of that by adding password-protection to the
+keystore. This is not be a breaking change: if a keystore has no password,
+there won’t be any new prompts. A user must choose to password-protect their
+keystore in order to benefit from the new behavior.
+
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[float]
+=== A new aggregation: `top_metrics`
+
+The new `top_metrics` aggregation "selects" a metric from a document according
+to a criteria on a given, different field. That criteria is currently being the
+largest or smallest "sort" value. It is fairly similar to `top_hits` in spirit
+but because it is more limited it is able to do its job using less memory and
+is often faster.
+
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+[float]
+=== Query speed-up for sorted queries on time-based indices
+
+Sorted queries run on time-based indices and interested in top documents only
+can be optimized by considering that the ranges of (document) timestamps in the
+shards don't overlap. This optimisation will rewrite the shard search requests
+based on the results available already from other shards when it can be
+determined that the query will not yield any result from the current shard;
+i.e. we know in advance that the bottom entry of the (sorted) result set after
+a partial merge is better than the values contained in this current shard.
```

Comment:
This one was admittedly a little hard for me to follow. I've made some changes to
try to help, but feel free to ignore if I've gotten something wrong.
```suggestion
We've optimized sorted, top-documents-only queries run on time-based indices.
{es} now considers the ranges of document timestamps in the shards, ensuring
they don't overlap. This optimization rewrites the shard search requests based
on results available already from other shards; i.e. we know the bottom entry of
the sorted results after a partial merge are better than the values contained in
this shard.
```

Reply:
The rephrasing is not accurate indeed, but the fact that it's not clear to you shows that it's not a great explanation. I've pushed another attempt, hopefully a bit clearer.

Reply:
The rephrase is much easier to read. Thanks, @bpintea.